js/ui/components/clubhouse.js

```diff
@@ -460,7 +482,11 @@ class ClubhouseNotificationBanner extends MessageTray.NotificationBanner {
             this.actor.x = endX;
         }
 
-        this.actor.y = margin;
+        if (this._position.atBottom) {
+            this.actor.y = monitor.y + monitor.height - this.actor.height - margin - taskbarHeight;
+        } else {
+            this.actor.y = margin;
+        }
```

Comment:
maybe you can create a helper function to calculate the value depending on the position. Or something else to avoid code duplication. I see the same code below.

Reply:
Yes the code duplication should be fixed before the final review. There is more code duplication in the current code that needs refactor, see `_slideOut()`.